src/value.rs

```diff
@@ -1534,10 +1527,10 @@ impl<'a> Set<'a> {
 
 impl<'a> Function<'a> {
     /// Create a function in the current execution context for a given callback.
-    pub fn new(isolate: &'a isolate::Isolate,
+    pub fn new<'b>(isolate: &'a isolate::Isolate,
```

Comment:
I'm not entirely sure what is needed to solve this, but I've reverted the lifetime and it is safe for now. It just requires setting an extra variable in scope.